Can't bypass DOSkey macro with leading space

<!-- 
🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨

I ACKNOWLEDGE THE FOLLOWING BEFORE PROCEEDING:
1. If I delete this entire template and go my own path, the core team may close my issue without further explanation or engagement.
2. If I list multiple bugs/concerns in this one issue, the core team may close my issue without further explanation or engagement.
3. If I write an issue that has many duplicates, the core team may close my issue without further explanation or engagement (and without necessarily spending time to find the exact duplicate ID number).
4. If I leave the title incomplete when filing the issue, the core team may close my issue without further explanation or engagement.
5. If I file something completely blank in the body, the core team may close my issue without further explanation or engagement.

All good? Then proceed!
-->

<!--
This bug tracker is monitored by Windows Terminal development team and other technical folks.

**Important: When reporting BSODs or security issues, DO NOT attach memory dumps, logs, or traces to Github issues**.
Instead, send dumps/traces to secure@microsoft.com, referencing this GitHub issue.

If this is an application crash, please also provide a Feedback Hub submission link so we can find your diagnostic data on the backend. Use the category "Apps > Windows Terminal (Preview)" and choose "Share My Feedback" after submission to get the link.

Please use this form and describe your issue, concisely but precisely, with as much detail as possible.

-->

# Environment

```none
Windows build number: [run `[Environment]::OSVersion` for powershell, or `ver` for cmd]
Windows Terminal version (if applicable):
Microsoft Windows [Version 10.0.18362.535]
Any other software? A DosKEY macro
```

# Steps to reproduce

<!-- A description of how to trigger this bug. -->
Create a DOSkey macro, e.g., 
`doskey dir=dir /b`
Run the command,
`dir`
then run the command (note _space in front_)
` dir`

# Expected behavior

<!-- A description of what you're expecting, possibly containing screenshots or reference material. -->
In the 2nd run, you should get the default "dir" listing. This is the case in classic console.

# Actual behavior

<!-- What's actually happening? -->
You get the results of "dir /b" again!

Still no dice :-(  bx rel built fine with one warning (spacing is straight copy from console):
`       "c:\devel\terminal\OpenConsole.sln" (Conhost\Host_EXE target) (1) ->
       "c:\devel\terminal\src\host\exe\Host.EXE.vcxproj.metaproj" (default targ
       et) (2) ->
       "c:\devel\terminal\src\host\proxy\Host.Proxy.vcxproj" (default target) (
       3) ->
       (Link target) ->
         LINK : warning LNK4266: missing load config symbol for image built wit
       h /GUARD [c:\devel\terminal\src\host\proxy\Host.Proxy.vcxproj]

    1 Warning(s)
    0 Error(s)`

The C runtimes are a common requirement, the version on my laptop was older, but installing the latest did not help. In comparing what was installed in Programs and Features, I found some discrepancies - perhaps .NET framework or its SDK is required? Or what about the version of the Windows SDK? Here are the diffs '<' is build system, '>' is test system:
`< Microsoft Visual C++ 2015-2019 Redistributable (x64) - 14.29.30139
> older version:                                         14.26.28720
> updated to latest:                                     14.31.31103

< Microsoft .NET Framework 4.8 SDK
<  Targeting Pack
<  Targeting Pack (ENU)
> None!
< Microsoft .NET SDK 5.0.406 (x64) from Visual Studio
> None!

< Windows Software Development Kit - Windows 10.0.14393.33 to 10.0.22000.194
>                                            10.0.17134.12 to 10.0.17763.132

< CLR types for SQL server 2019  CTP2.2
> CLR types for SQL server vNext CTP 1.6                          
< Sync Framework 2.0 Core and Provider
> Same`

--- a/src/host/alias.cpp
+++ b/src/host/alias.cpp
@@ -818,18 +818,6 @@ void Alias::s_ClearCmdExeAliases()
 }
 
 // Routine Description:
-// - Trims leading spaces off of a string
-// Arguments:
-// - str - String to trim
-void Alias::s_TrimLeadingSpaces(std::wstring& str)
-{
-    // Erase from the beginning of the string up until the first
-    // character found that is not a space.
-    str.erase(str.begin(),
-              std::find_if(str.begin(), str.end(), [](wchar_t ch) { return !std::iswspace(ch); }));
-}
-
-// Routine Description:
 // - Trims trailing \r\n off of a string
 // Arguments:
 // - str - String to trim
@@ -1147,9 +1135,6 @@ std::wstring Alias::s_MatchAndCopyAlias(const std::wstring& sourceText,
 
     // Trim trailing \r\n off of sourceCopy if it has one.
     s_TrimTrailingCrLf(sourceCopy);
-
-    // Trim leading spaces off of sourceCopy if it has any.
-    s_TrimLeadingSpaces(sourceCopy);
 
     // Check if we have an EXE in the list that matches the request first.
     auto exeIter = g_aliasData.find(exeName);

--- a/src/host/alias.h
+++ b/src/host/alias.h
@@ -29,7 +29,6 @@ public:
                                             size_t& lineCount);
 
 private:
-    static void s_TrimLeadingSpaces(std::wstring& str);
     static void s_TrimTrailingCrLf(std::wstring& str);
     static std::deque<std::wstring> s_Tokenize(const std::wstring& str);
     static std::wstring s_GetArgString(const std::wstring& str);

--- a/src/host/ut_host/AliasTests.cpp
+++ b/src/host/ut_host/AliasTests.cpp
@@ -340,10 +340,10 @@ class AliasTests
         auto rgwchTargetBefore = std::make_unique<wchar_t[]>(cchTarget);
         wcscpy_s(rgwchTargetBefore.get(), cchTarget, rgwchTarget.get());
         size_t cbTargetUsed = 0;
-        const auto cbTargetUsedExpected = cbTarget;
+        const auto cbTargetUsedBefore = cbTargetUsed;
 
         DWORD dwLines = 0;
-        const auto dwLinesExpected = dwLines + 1;
+        const auto dwLinesBefore = dwLines;
 
         // Register the correct alias name before we try.
         std::wstring exe(L"exe.exe");
@@ -351,9 +351,7 @@ class AliasTests
         std::wstring target(L"someTarget");
         Alias::s_TestAddAlias(exe, source, target);
 
-        auto targetExpected = target + L"\r\n";
-
-        // We should be able to match through the leading spaces. They should be stripped.
+        // Leading spaces should bypass the alias. This should not match anything.
         Alias::s_MatchAndCopyAliasLegacy(pwszSource,
                                          cbSource,
                                          rgwchTarget.get(),
@@ -362,9 +360,9 @@ class AliasTests
                                          exe,
                                          dwLines);
 
-        VERIFY_ARE_EQUAL(cbTargetUsedExpected, cbTargetUsed, L"No target bytes should be used.");
-        VERIFY_ARE_EQUAL(String(targetExpected.data(), gsl::narrow<int>(targetExpected.size())), String(rgwchTarget.get(), cchTarget), L"Target string should match expected.");
-        VERIFY_ARE_EQUAL(dwLinesExpected, dwLines, L"Line count be updated to 1.");
+        VERIFY_ARE_EQUAL(cbTargetUsedBefore, cbTargetUsed, L"No bytes should be used if nothing was found.");
+        VERIFY_ARE_EQUAL(String(rgwchTargetBefore.get(), cchTarget), String(rgwchTarget.get(), cchTarget), L"Target string should be unmodified.");
+        VERIFY_ARE_EQUAL(dwLinesBefore, dwLines, L"Line count should pass through.");
     }
 
     TEST_METHOD(TrimTrailing)

